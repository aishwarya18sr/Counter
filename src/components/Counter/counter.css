.counterContainer {
    display: flex;
    flex-direction: column;
    width:100vw;
    height:100vh;
    justify-content: center;
    align-items: center;
}

.counterValue {
    font-size: 60px;
}

.counterButtonContainer {
    display: flex;
    flex-direction: row;
    gap: 40px;
    height: 60px;
}

.counterButton {
    display:flex;
    align-items: center;
    justify-content: center;
    width: 60px;
    font-size: 40px;
    cursor: pointer;
}