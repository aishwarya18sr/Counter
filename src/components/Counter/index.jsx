import React, { useState } from "react";
import "./counter.css";

const Counter = () => {
  const [counter, setCounter] = useState(0);
  return (
    <div className="counterContainer">
      <p className="counterValue">{counter}</p>
      <div className="counterButtonContainer">
        <button
          type="button"
          className="counterButton"
          onClick={() => setCounter(counter + 1)}
        >
          +
        </button>
        <button
          type="button"
          className="counterButton"
          onClick={() => {
            if (counter > 1) {
              setCounter(counter - 1);
            } else {
              setCounter(0);
            }
          }}
        >
          -
        </button>
      </div>
    </div>
  );
};

export default Counter;
