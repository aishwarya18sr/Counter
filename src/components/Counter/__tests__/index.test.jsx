import { fireEvent, render } from "@testing-library/react";
import React from "react";
import Counter from "..";

describe("Counter", () => {
  it("should display 0 when renders for the first time", () => {
    const { queryByText } = render(<Counter />);
    expect(queryByText(0)).toBeTruthy();
  });
  it("should increase the counter value by 1 when increment button is clicked", () => {
    const { queryByText } = render(<Counter />);
    const incrementButton = queryByText("+");
    expect(queryByText(1)).toBeFalsy();
    fireEvent.click(incrementButton);
    expect(queryByText(1)).toBeTruthy();
  });
  it("should decrease the counter value by 1 when decrement button is clicked and counter value is greater than 0", () => {
    const { queryByText } = render(<Counter />);
    const incrementButton = queryByText("+");
    fireEvent.click(incrementButton);
    const decrementButton = queryByText("-");
    expect(queryByText(0)).toBeFalsy();
    fireEvent.click(decrementButton);
    expect(queryByText(0)).toBeTruthy();
  });
  it("should have counter value as 0 when decrement button is clicked and the counter value is already 0", () => {
    const { queryByText } = render(<Counter />);
    const decrementButton = queryByText("-");
    fireEvent.click(decrementButton);
    expect(queryByText(0)).toBeTruthy();
  });
  it("should render correctly", () => {
    const { asFragment } = render(<Counter />);
    expect(asFragment()).toMatchSnapshot();
  });
});
