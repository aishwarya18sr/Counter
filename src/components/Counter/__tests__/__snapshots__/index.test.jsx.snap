// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter should render correctly 1`] = `
<DocumentFragment>
  <div
    class="counterContainer"
  >
    <p
      class="counterValue"
    >
      0
    </p>
    <div
      class="counterButtonContainer"
    >
      <button
        class="counterButton"
        type="button"
      >
        +
      </button>
      <button
        class="counterButton"
        type="button"
      >
        -
      </button>
    </div>
  </div>
</DocumentFragment>
`;
